### DISCUSSION TITLE
How to read keyboard events using only the core libraries?

### DISCUSSION THREAD

**Discussion by kevinmatthes** [odin-lang/Odin#3130] [Q&A] [answered]
First of all, thank you for your work on this great language!

In order to learn Odin language, I would like to create myself a small terminal text editor.  Therefore, I need to read keyboard events.  Since my main objective is to become familiar with the standard library, I was wondering whether it might be possible to use only `core` instead of binding third-party libraries, such as ncurses.  I already did some research on this topic but I did not succeed, yet; I am sure that I oversaw something obvious...

Does anyone have some idea how to read keyboard events using only the core libraries?

**Comment by sumofat** [comment, marked as answer]
Core library for that kind of thing does not really exist.  You would need to create your own input library using the available OS specific options and bindings.  
Included with ODIN are the vendor libraries and you can use SDL or raylib glfw or others to get keyboard events in a very easy way. 
Otherwise I believe you would have to do it the normal platform api specific way.  
For example on windows using VkKeyScanExA and GetKeyboardState apis which I believe are available in the core library.
You would need to do that for each platform.  If thats not what your looking for than use the previously mentioned platform abstraction layers SDL,GLFW or raylib etc..

**Reply by kevinmatthes** [reply]
Thank you for your help!  I will give it a try this way.
